tests/director/tests/application_templates_subscription_test.go

```diff
@@ -0,0 +1,110 @@
+package tests
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
+	"github.com/kyma-incubator/compass/tests/pkg/certs/certprovider"
+	"github.com/kyma-incubator/compass/tests/pkg/fixtures"
+	"github.com/kyma-incubator/compass/tests/pkg/gql"
+	"github.com/kyma-incubator/compass/tests/pkg/subscription"
+	"github.com/kyma-incubator/compass/tests/pkg/tenant"
+	"github.com/kyma-incubator/compass/tests/pkg/testctx"
+	testingx "github.com/kyma-incubator/compass/tests/pkg/testing"
+	"github.com/kyma-incubator/compass/tests/pkg/token"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSubscriptionApplicationTemplateFlow(stdT *testing.T) {
+	t := testingx.NewT(stdT)
+	t.Run("Subscribe tenant to Application flow", func(t *testing.T) {
+		ctx := context.Background()
+		subscriptionProviderSubaccountID := conf.TestProviderSubaccountID
+		subscriptionConsumerSubaccountID := conf.TestConsumerSubaccountID
+		subscriptionConsumerTenantID := conf.TestConsumerTenantID
+
+		// Prepare provider external client certificate and secret and Build graphql director client configured with certificate
+		providerClientKey, providerRawCertChain := certprovider.NewExternalCertFromConfig(t, ctx, conf.ExternalCertProviderConfig)
+		directorCertSecuredClient := gql.NewCertAuthorizedGraphQLClientWithCustomURL(conf.DirectorExternalCertSecuredURL, providerClientKey, providerRawCertChain, conf.SkipSSLValidation)
+
+		// Create Application Template
+		appTemplateName := createAppTemplateName("app-template-name-subscription")
```

Comment:
With e2e test the issue described above can be tested if you create more than one AppTemplate in the same region and verify that subscription is removed on unsubscribe, the expectation would be that the test will be unstable - will pass half of the times. To make it more unstable, you may create 10 or 20 templates in the same region as part of the test instead 2.